mllib/src/main/scala/org/apache/spark/ml/util/Instrumentation.scala

```diff
@@ -85,9 +86,27 @@ private[spark] class Instrumentation[E <: Estimator[_]] private (
   }
 
   /**
+   * Log params for tuning estimators that search for the best params by wrapping another
+   * estimator and using metrics computed by an evaluator.
+   *
+   * @param estimator the inner estimator called by the tuning estimator
+   * @param estimatorParamMaps different params tried by the tuning estimator
+   * @param evaluator evaluator used to compute the metric for each estimator param value
+   */
+  def logTuningParams(
+    estimator: Estimator[_],
+    estimatorParamMaps: Array[ParamMap],
+    evaluator: Evaluator): Unit = {
+    log(compact(render(map2jvalue(Map[String, JValue](
+      "estimator" -> estimator.getClass.getSimpleName,
+      "evaluator" -> evaluator.getClass.getSimpleName,
+      "numModels" -> estimatorParamMaps.length)))))
+  }
+
+  /**
    * Logs the value with customized name field.
    */
-  def logNamedValue(name: String, num: Long): Unit = {
+  def logNamedValue(name: String, num: JValue): Unit = {
```

Comment:
Since this class doesn't expose json4s APIs, let's stick with basic types in the public API.  Just do String for now (since the Long version is not yet used).